contrib/linux/actions/wait_for_ssh.yaml

```diff
@@ -1,23 +1,34 @@
 ---
   name: "wait_for_ssh"
   runner_type: "python-script"
-  description: "Wait for SSH"
+  description: "Action which waits for a SSH server to become accessible. By default if no credentials are explicitly provided, this action will try to authenticate using system user username and password."
   enabled: true
   entry_point: "wait_for_ssh.py"
-  parameters: 
-    keyfile: 
-      default: "/home/stanley/.ssh/stanley_rsa"
+  parameters:
+    hostname:
+      description: "Remote hostname"
+      type: "string"
       required: true
-    username: 
-      description: "Command line arguments"
+    port:
+      description: "Remote SSH port."
+      type: "integer"
+      default: 22
+    username:
+      description: "Username used to authenticate."
       required: true
       default: "stanley"
-    hostname: 
-      type: "string"
-      required: true
-    ssh_timeout: 
+    password:
+      description: "Password used to authenticate."
+      required: false
+    keyfile:
+      description: "SSH key file used to authenticate."
+      required: false
+    # TODO: We should really rename this to sleep delay or similar
+    ssh_timeout:
```

Comment:
I really want to rename this argument since current name is deceiving and incorrect.

Right now it represents a sleep delay between each connection attempt and not a SSH connection timeout.

I was thinking about actually utilizing this parameter for SSH connection timeout and introducing a new `sleep_delay` (or similar) parameter which represents a sleep delay between each SSH connection attempt.

What do others think?